src/queue.rs

```diff
@@ -0,0 +1,148 @@
+mod job;
+pub use crate::entity::queue::{JobResult, JobStatus};
+pub use job::*;
+
+use crate::{
+    entity::queue::{ActiveModel, Entity, Model},
+    ApiConfig, Db,
+};
+use sea_orm::{
+    ActiveModelTrait, ConnectionTrait, DatabaseBackend, DatabaseTransaction, DbErr, EntityTrait,
+    IntoActiveModel, Set, Statement, TransactionTrait,
+};
+use std::{ops::Range, sync::Arc, time::Duration};
+use time::OffsetDateTime;
+use tokio::{task::JoinSet, time::sleep};
+
+/*
+These configuration variables may eventually be useful to put on ApiConfig
+*/
+const MAX_RETRY: i32 = 5;
+const QUEUE_CHECK_INTERVAL: Range<u64> = 10..20;
+const QUEUE_WORKER_COUNT: u8 = 2;
+
+fn schedule_based_on_failure_count(failure_count: i32) -> Option<OffsetDateTime> {
+    if failure_count >= MAX_RETRY {
+        None
+    } else {
+        Some(
+            OffsetDateTime::now_utc()
+                + Duration::from_millis(
+                    4000_u64.pow(failure_count.try_into().unwrap()) + fastrand::u64(0..15000),
+                ),
+        )
+    }
+}
+
+pub async fn one(db: &Db, job_state: &SharedJobState) -> Result<Option<Model>, DbErr> {
```

Comment:
One what? Does this dequeue something from the queue? Maybe a doccomment to explain what this does?